partner_communication/models/communication_job.py

```diff
@@ -825,55 +825,48 @@ def print_letter(self, print_name, **print_options):
-        config_lang = printer_configs.filtered(lambda c: c.lang_id.code == lang)
-        if len(config_lang) > 1:
-            raise UserError(_(
-                "Cannot print multiple communication types at the same time."))
-
-        # The trays are defined at 3 places, get options in order of priority
-        # Output tray - partner.communication.job
-        if len(self) == 1 and self.printer_output_tray_id.system_name:
-            output_tray = self.printer_output_tray_id.system_name
-        # Output tray - partner.communication.config
-        elif config_lang.printer_output_tray_id.system_name:
-            output_tray = config_lang.printer_output_tray_id.system_name
-        # Output tray - ir.actions.report
-        elif report.printer_output_tray_id.system_name:
-            output_tray = report.printer_output_tray_id.system_name
-        else:
-            output_tray = False
-        # Handle input tray similarly to output tray
-        if config_lang.printer_input_tray_id.system_name:
-            input_tray = config_lang.printer_input_tray_id.system_name
-        elif report.printer_input_tray_id.system_name:
-            input_tray = report.printer_input_tray_id.system_name
-        else:
-            input_tray = False
+        print_options["doc_format"] = report.report_type
+        print_options["action"] = behaviour["action"]
+
+        # The get the print options in the following order of priority:
+        # - partner.communication.job (only for output bin)
+        # - partner.communication.config
+        # - ir.actions.report (behaviour: which can be specific for the user currently logged in)
+        def get_first(source):
+            return next((v for v in source if v), False)
+
+        print_options["output_tray"] = get_first((
+            self[0:].printer_output_tray_id.system_name,
+            config_lang.printer_output_tray_id.system_name,
+            behaviour["output_tray"]
+        ))
+        print_options["input_tray"] = get_first((
+            config_lang.printer_input_tray_id.system_name,
+            behaviour["input_tray"]
+        ))
 
-        if behaviour["action"] != "client" and printer:
-            # Get pdf should directly send it to the printer if report
-            # is correctly configured.
+        printer = behaviour["printer"].with_context(lang=lang)
+        if behaviour["action"] == "server" and printer:
+            # Get pdf should directly send it to the printer
             to_print = report.render_qweb_pdf(self.ids)
-            printer.print_document(
-                report.report_name, to_print[0],
-                doc_format=report.report_type,
-                action=behaviour['action'],
-                input_tray=input_tray,
-                output_tray=output_tray,
-                **print_options
-            )
-        return output_tray
+            printer.print_document(report.report_name, to_print[0], **print_options)
+
+        return print_options
```

Comment:
By changing the return type, don't you break anything else in the code? Did you check all calls to this method so that they continue to behave as expected?

Reply:
Yes, I've checked with the PyCharm Odoo plugin and the only part that were using the output I've changed it